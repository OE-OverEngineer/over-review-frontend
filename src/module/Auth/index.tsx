import React from 'react';

import { LockOutlined, UserOutlined } from '@ant-design/icons';
import { Button, Checkbox, Form, Input, Layout } from 'antd';
import { Content, Footer } from 'antd/lib/layout/layout';

import Star from 'common/assets/images/star.svg';
import Navbar from 'common/components/Navbar';

const Auth: React.FC = () => {
  return (
    <Layout className="max-h-screen  h-screen">
      <Star className="absolute h-full w-full z-0" />

      <Navbar />
      <Content>
        <section className="max-w-screen-2xl m-auto mt-28">
          <p className=" text-3xl text-center"> </p>
          <Form
            name="loginForm"
            className="max-w-lg m-auto bg-white pt-3 pb-4 rounded-3xl "
            layout="vertical"
            initialValues={{ remember: true }}
            wrapperCol={{ offset: 4, span: 16 }}
            labelCol={{ offset: 4, span: 16 }}
            autoComplete="off">
            <p className=" text-primary-default text-4xl text-center mt-4 ">Sign In</p>
            <Form.Item
              label="Username"
              name="username"
              rules={[{ required: true, message: 'Please input your username!' }]}>
              <Input
                prefix={<UserOutlined style={{ color: '#7433FF' }} />}
                className="rounded-lg"
              />
            </Form.Item>
            <Form.Item
              label="Password"
              name="password"
              rules={[{ required: true, message: 'Please input your password!' }]}>
              <Input.Password
                prefix={<LockOutlined style={{ color: '#7433FF' }} />}
                className="rounded-lg"
              />
            </Form.Item>
            <Form.Item name="remember" valuePropName="checked">
              <Checkbox className="text-primary-default">Remember me</Checkbox>
            </Form.Item>
            <Form.Item className="text-center">
              <Button
                type="primary"
                htmlType="submit"
                shape="round"
                style={{ width: '128px' }}>
                Login
              </Button>
            </Form.Item>
            <p className=" text-purple-400 text-center">
              Create an account<span className="text-gray-400"> to join OVER review</span>
            </p>
          </Form>
        </section>
      </Content>
      <Footer className="text-center">Over Review Â©2021 Created by Over Engineer</Footer>
    </Layout>
  );
};

export default Auth;
