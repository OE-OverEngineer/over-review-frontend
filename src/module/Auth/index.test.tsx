import React from 'react';

import { Button, Input } from 'antd';
import { shallow } from 'enzyme';

import Auth from '.';

describe('Auth page', () => {
  it('Auth should be defined', () => {
    const wrap = shallow(<Auth />);
    wrap.find(Input).simulate('change', { target: { value: 'test' } });
    wrap.find(Button).simulate('click');
    expect(wrap.find('div[role="alert"]').debug()).toEqual(
      '<div role="alert" class="ant-form-item-explain-error" style="">Please input your password!</div>',
    );

    // expect(toJson(wrap)).toMatchSnapshot();
  });
});
