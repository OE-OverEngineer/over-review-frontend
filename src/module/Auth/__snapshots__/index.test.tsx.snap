// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth page Auth should be defined 1`] = `
<Layout
  className="min-h-screen h-full font-poppins"
>
  <Navbar />
  <Content
    className="z-10"
  >
    <section
      className="max-w-screen-2xl m-auto mt-28"
    >
      <ForwardRef(InternalForm)
        autoComplete="off"
        className="max-w-lg m-auto bg-white pt-3 pb-4 rounded-3xl "
        initialValues={
          Object {
            "remember": true,
          }
        }
        labelCol={
          Object {
            "offset": 4,
            "span": 16,
          }
        }
        layout="vertical"
        name="loginForm"
        onFinish={[Function]}
        wrapperCol={
          Object {
            "offset": 4,
            "span": 16,
          }
        }
      >
        <p
          className=" text-primary-default text-4xl text-center mt-4 "
        >
          Sign In
        </p>
        <FormItem
          className="hidden-required"
          label="Username"
          name="username"
          rules={
            Array [
              Object {
                "message": "Please input your username!",
                "required": true,
              },
            ]
          }
        >
          <Input
            className="rounded-lg"
            prefix={
              <ForwardRef(UserOutlined)
                style={
                  Object {
                    "color": "#7433FF",
                  }
                }
              />
            }
            type="text"
          />
        </FormItem>
        <FormItem
          className="hidden-required"
          label="Password"
          name="password"
          rules={
            Array [
              Object {
                "message": "Please input your password!",
                "required": true,
              },
            ]
          }
        >
          <Password
            action="click"
            className="rounded-lg"
            iconRender={[Function]}
            prefix={
              <ForwardRef(LockOutlined)
                style={
                  Object {
                    "color": "#7433FF",
                  }
                }
              />
            }
            visibilityToggle={true}
          />
        </FormItem>
        <FormItem
          name="remember"
          valuePropName="checked"
        >
          <Checkbox
            className="text-primary-default"
          >
            Remember me
          </Checkbox>
        </FormItem>
        <FormItem
          className="text-center"
        >
          <Button
            htmlType="submit"
            shape="round"
            style={
              Object {
                "width": "128px",
              }
            }
            type="primary"
          >
            Login
          </Button>
        </FormItem>
        <div
          className=" text-purple-400 text-center"
        >
          <Link
            href="/register"
          >
            Create an account
          </Link>
          <span
            className="text-gray-400"
          >
             to join OVER review
          </span>
        </div>
      </ForwardRef(InternalForm)>
    </section>
    <ToastContainer
      autoClose={5000}
      closeButton={[Function]}
      closeOnClick={true}
      draggable={true}
      draggableDirection="x"
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      role="alert"
      rtl={false}
      theme="light"
      transition={[Function]}
    />
  </Content>
  <Footer
    className="text-center"
  >
    Over Review Â©2021 Created by Over Engineer
  </Footer>
</Layout>
`;
